Sources/Sentry/SentrySessionReplayIntegration.m

```diff
@@ -98,58 +108,49 @@ - (void)startWithOptions:(SentryReplayOptions *)replayOptions
-             displayLinkWrapper:[[SentryDisplayLinkWrapper alloc] init]];
-
-        [self.sessionReplay
-                  start:SentryDependencyContainer.sharedInstance.application.windows.firstObject
-            fullSession:[self shouldReplayFullSession:replayOptions.sessionSampleRate]];
-
-        [NSNotificationCenter.defaultCenter addObserver:self
-                                               selector:@selector(stop)
-                                                   name:UIApplicationDidEnterBackgroundNotification
-                                                 object:nil];
-
-        [NSNotificationCenter.defaultCenter addObserver:self
-                                               selector:@selector(resume)
-                                                   name:UIApplicationWillEnterForegroundNotification
-                                                 object:nil];
-
-        [SentryGlobalEventProcessor.shared
-            addEventProcessor:^SentryEvent *_Nullable(SentryEvent *_Nonnull event) {
-                [self.sessionReplay captureReplayForEvent:event];
-                return event;
-            }];
+    NSURL *docs =
+        [NSURL fileURLWithPath:[SentryDependencyContainer.sharedInstance.fileManager sentryPath]];
+    docs = [docs URLByAppendingPathComponent:SENTRY_REPLAY_FOLDER];
+    NSString *currentSession = [NSUUID UUID].UUIDString;
+    docs = [docs URLByAppendingPathComponent:currentSession];
+
+    if (![NSFileManager.defaultManager fileExistsAtPath:docs.path]) {
+        [NSFileManager.defaultManager createDirectoryAtURL:docs
+                               withIntermediateDirectories:YES
+                                                attributes:nil
+                                                     error:nil];
     }
+
+    SentryOnDemandReplay *replayMaker = [[SentryOnDemandReplay alloc] initWithOutputPath:docs.path];
+    replayMaker.bitRate = replayOptions.replayBitRate;
+    replayMaker.cacheMaxSize
+        = (NSInteger)(shouldReplayFullSession ? replayOptions.sessionSegmentDuration
+                                              : replayOptions.errorReplayDuration);
+
+    self.sessionReplay = [[SentrySessionReplay alloc]
+           initWithSettings:replayOptions
+           replayFolderPath:docs
+         screenshotProvider:screenshotProvider
+                replayMaker:replayMaker
+        breadcrumbConverter:breadcrumbConverter
+               touchTracker:_touchTracker
+               dateProvider:SentryDependencyContainer.sharedInstance.dateProvider
+                     random:SentryDependencyContainer.sharedInstance.random
+         displayLinkWrapper:[[SentryDisplayLinkWrapper alloc] init]];
+
+    [self.sessionReplay
+              start:SentryDependencyContainer.sharedInstance.application.windows.firstObject
+        fullSession:[self shouldReplayFullSession:replayOptions.sessionSampleRate]];
+
+    [NSNotificationCenter.defaultCenter addObserver:self
```

Comment:
`m`: Some other code in this class already uses the `NotificationCenterWrapper`. It would be great to stick to one of the two.